source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs

```diff
@@ -103,37 +108,30 @@ public void UpdateCharge(ChargePeriod newChargePeriod)
         {
             if (newChargePeriod == null) throw new ArgumentNullException(nameof(newChargePeriod));
 
-            RemoveAllPeriodsFromNewPeriodStart(newChargePeriod);
-            HandleAnyOverlappingPeriod(newChargePeriod);
+            RemoveAllSubsequentPeriods(newChargePeriod.StartDateTime);
+            StopExistingPeriod(newChargePeriod.StartDateTime);
             _periods.Add(newChargePeriod);
             Validate();
         }
 
-        private void HandleAnyOverlappingPeriod(ChargePeriod newChargePeriod)
+        private void StopExistingPeriod(Instant dateTime)
         {
             var previousPeriod = _periods
                 .SingleOrDefault(p =>
-                p.EndDateTime > newChargePeriod.StartDateTime &&
-                p.StartDateTime < newChargePeriod.StartDateTime);
+                    p.EndDateTime > dateTime &&
+                    p.StartDateTime < dateTime);
 
             if (previousPeriod != null)
             {
-                var newPreviousPeriod = new ChargePeriod(
-                    previousPeriod.Id,
-                    previousPeriod.Name,
-                    previousPeriod.Description,
-                    previousPeriod.VatClassification,
-                    previousPeriod.TransparentInvoicing,
-                    previousPeriod.StartDateTime,
-                    newChargePeriod.StartDateTime);
+                var newPreviousPeriod = ChargePeriodFactory.CreateFromExistingPeriodWithNewEndDate(previousPeriod, dateTime);
```

Comment:
I personally like this approach:
```suggestion
                var newPreviousPeriod = previousPeriod.WithEndDate(dateTime);
```